src/Modules/UngradedModules/DirectSum.jl

```diff
@@ -19,34 +19,23 @@ function direct_product(F::FreeMod{T}...; task::Symbol = :prod) where {T}
   if all_graded
     G.d = vcat([f.d for f in F]...)
   end
-  G.S = []
-  for i = 1:length(F)
-    s = "("
-    for j=1:i-1
-      s *= "0, "
-    end
-    e = ""
-    if i<length(F)
-      e*=", "
-    end
-    for j=i+1:length(F)-1
-      e *= "0, "
-    end
-    if i<length(F)
-      e *= "0"
-    end
-    e*=")"
-
-    for t = F[i].S
-      push!(G.S, Symbol(s*string(t)*e))
-    end
-  end
```

Comment:
This has shown to eat a lot of resources for large direct sums. I tried to address this by using `join` and friends below to eventually allow for more compiler optimization.